### PR TITLE
merge build.yml and fmt.yml

### PR DESCRIPTION
**Pull request by henry40408**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [x] Other, please describe: **GitHub workflow configuration**

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

This pull request...

1. merges `build.yml` and `fmt.yml`
2. replaces `hecrj/setup-rust-action@v1` with `actions-rs/cargo@v1`
3. removes the following steps because `serde` feature is not used in `Cargo.toml`:
    - Build with serde enabled
    - Build tests with serde enabled
    - Run tests with serde enabled

### CODE REVIEW

**Comment by amrbashir** [comment]
I am fine with merging fmt.yml into build.yml but it needs to be its own job.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,26 +23,27 @@ jobs:
...
-      RUSTFLAGS: "-C debuginfo=0"
-      OPTIONS: ${{ matrix.platform.options }}
-      FEATURES: ${{ format(',{0}', matrix.platform.features ) }}
-      CMD: ${{ matrix.platform.cmd }}
+      RUST_BACKTRACE: 1 # rustc option
+      CARGO_INCREMENTAL: 0 # cargo option
+      RUSTFLAGS: "-C debuginfo=0" # rustc option
+      OPTIONS: ${{ matrix.platform.options }} # as arguments
+      FEATURES: ${{ format(',{0}', matrix.platform.features ) }} # as arguments
+      CMD: ${{ matrix.platform.cmd }} # as arguments
```
Comment (amrbashir) [inline .github/workflows/build.yml]: The comments are not needed

**Review by amrbashir — CHANGES_REQUESTED** [review]
Can you also rename `build.yml` to `ci.yml`

File: .github/workflows/build.yml (hunk 1 above)
Comment (henry40408) [reply .github/workflows/build.yml]: removed in 0b5ddd4

File: Cargo.toml [hunk 2]
```diff
@@ -16,7 +16,7 @@ documentation = "https://docs.rs/tao"
 categories = [ "gui" ]
 
 [package.metadata.docs.rs]
-features = [ "serde", "tray", "dox" ]
```
Comment (amrbashir) [inline Cargo.toml]: why was this removed?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -81,18 +85,16 @@ jobs:
...
-      - name: Build tests with serde enabled
-        shell: bash
-        run: cargo $CMD test --no-run --verbose --target ${{ matrix.platform.target }} $OPTIONS --features serde,$FEATURES
-
-      - name: Run tests with serde enabled
-        shell: bash
-        if: (
-          !contains(matrix.platform.target, 'android') &&
-          !contains(matrix.platform.target, 'ios'))
-        run: cargo $CMD test --verbose --target ${{ matrix.platform.target }} $OPTIONS --features serde,$FEATURES
```
Comment (amrbashir) [inline .github/workflows/ci.yml]: why was these removed too?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -63,6 +64,9 @@ jobs:
         if: contains(matrix.platform.target, 'android')
         run: cargo install cargo-apk
 
+      - name: Download dependencies
+        run: cargo $CMD fetch
+
```
Comment (amrbashir) [inline .github/workflows/ci.yml]: dependencies will be downloaded when building anyways? is there a reason to separate this into its own step?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (amrbashir) [reply .github/workflows/ci.yml]: looks like we also don't run tests with `tray` feature enabled, maybe we should add that too.

**Review by amrbashir — COMMENTED** [review]
Sorry, I didn't notice these in the original review

File: .github/workflows/ci.yml (hunk 3 above)
Comment (wusyong) [reply .github/workflows/ci.yml]: Maybe we can have a test with full features and one without.
It's not practical to run every feature in a single test job.

Comment (amrbashir) [reply .github/workflows/ci.yml]: yeah that sounds better

File: Cargo.toml (hunk 2 above)
Comment (henry40408) [reply Cargo.toml]: add back on commit 71bb9a2

File: .github/workflows/ci.yml (hunk 4 above)
Comment (henry40408) [reply .github/workflows/ci.yml]: removed on commit 71bb9a2

File: .github/workflows/ci.yml (hunk 3 above)
Comment (henry40408) [reply .github/workflows/ci.yml]: I added a new `platform` in matrix: `ubuntu-with-all-features`, which includes all features described in `Cargo.toml`

Comment (amrbashir) [reply .github/workflows/ci.yml]: we need to test the tray on all platforms, not just ubuntu, so you should just re-add these 3 steps and add tray to it

**Review by amrbashir — APPROVED** [review]
Thank you
